graphqlUtils.go

```diff
@@ -0,0 +1,204 @@
+package resource
+
+import (
+	"github.com/graphql-go/graphql/language/ast"
+	"github.com/graphql-go/graphql/language/parser"
+	"github.com/graphql-go/graphql/language/printer"
+)
+
+func combineSchemas(oldSDL string, newSDL string, resolvers []Resolver) (string, error) {
+	oldSchema, err := parser.Parse(parser.ParseParams{
+		Source: string(oldSDL),
+		Options: parser.ParseOptions{
+			NoLocation: false,
+			NoSource:   true,
+		},
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	newSchema, err := parser.Parse(parser.ParseParams{
+		Source: string(newSDL),
+		Options: parser.ParseOptions{
+			NoLocation: false,
+			NoSource:   true,
+		},
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	// Ensure that resolved fields are in the schema
+	for _, resolver := range resolvers {
+		existingType := getTypeFromSchema(oldSchema, resolver.TypeName)
+
+		if existingType == nil {
+			continue
+		}
+
+		existingObjectType := existingType.(*ast.ObjectDefinition)
```

Comment:
If the assertion is false it will panic.

Reply:
I was working with an assumption here that resolvers can return only object types.. but that's obviously not true. I'll fix

Reply:
Oh wait it is.. resolvers can be only attached to object types. I'll put in some checks so it doesn't panic though.

Reply:
Fixed